CruzHacks/cruzhacks-2024-website
# Event Announcements

# Scope

Implemented the [Event Announcements](https://github.com/CruzHacks/cruzhacks-2024-website/issues/44); admins can post events and hackers can view them.

# Implementation

Activated the Realtime Database for the project and used its associated emulator. [You can find last years code here.](https://github.com/CruzHacks/cruzhacks-2023-live-site/blob/development/src/utils/firebase.ts)

Added a modal to the admin portal for creating an announcement. Used React Toastify notifications to indicate success and failure. [You can find last years code here.](https://github.com/CruzHacks/cruzhacks-2023-website/blob/development/src/views/Portal/Admin/Dashboard/index.tsx)

Added a component to the hacker portal for showing the announcements. Listed in decreasing order; newer announcements displayed on top of older ones. [You can find last years code here.](https://github.com/CruzHacks/cruzhacks-2023-live-site/blob/development/src/views/Home/Notifications.tsx)

# Images/Important Notes (optional):

Added React Toastify.

# How to Test

Admin
* Make an announcement with a description. This should succeed and appear in the emulator on port 9000
* Make an announcement without a description. This should give you a warning indicating you need a description for the announcement.

Hacker
* After making an announcement as a Admin, a hacker should be able to see them.
